Feature/etm keyedvectors embeddings input file

lffloyd opened the pull request:
Resolves #32, #37.

Adds support for original-formatted word2vec and [```gensim.models.KeyedVectors```](https://radimrehurek.com/gensim/models/keyedvectors.html)-formatted embeddings files for ETM model training.
Updates some details on the ```CONTRIBUTING``` section of the documentation.

silviatti commented on octis/models/ETM.py:
```diff
@@ -16,7 +16,7 @@ class ETM(BaseETM):
                  activation='relu', dropout=0.5, lr=0.005, optimizer='adam', batch_size=128, clip=0.0,
                  wdecay=1.2e-6, bow_norm=1, device='cpu', top_word=10, train_embeddings=True, embeddings_path=None,
-                 use_partitions=True):
+                 embeddings_type='pickle', binary_embeddings=True, use_partitions=True):
```
Can you add the comments to explain the parameters "embeddings_type" and "binary_embeddings"? Including the type and default values (I'll take care of the others parameters).

lffloyd replied:
Done

silviatti commented on octis/models/base_etm.py:
```diff
@@ -45,24 +46,35 @@ def preprocess(vocab2id, train_corpus, test_corpus=None, validation_corpus=None)
+                embeddings[i] = np.random.normal(scale=0.6, size=(self.hyperparameters['embedding_size'],))
+        self.embeddings = torch.from_numpy(embeddings).to(self.device)
+
+    def _load_word_vectors(self, embeddings_path, embeddings_type, binary_embeddings=True):
```
Can you add the comments to explain the parameters of the method? Including the type and default values

lffloyd replied:
Done

silviatti reviewed (changes requested):
Thank you for your work! :) I just added a few comments to address before merging

lffloyd commented:
> Thank you for your work! :) I just added a few comments to address before merging

Thanks for the feedback, @silviatti! I've added docstrings for the changed methods, at your request.

lffloyd commented:
Hi @silviatti . Updated the PR with [a commit](https://github.com/MIND-Lab/OCTIS/pull/36/commits/086d7ebe6aace2908295b1488580b5e47809f141) adding support for headerless original word2vec embeddings file in textual format. This feature provides support for the Dieng's pretrained embeddings found [here](https://bitbucket.org/diengadji/embeddings/src/master/), as requested by [cayaluke's comment](https://github.com/MIND-Lab/OCTIS/issues/32#issuecomment-963716020) on #32. Previously, my PR supported only regular files including the header.

silviatti reviewed (approved):
Thank you again for your work and patience :) I'm going to merge the changes and I'll soon make the new release.

Silvia
